service/src/test/java/org/jboss/sbomer/service/test/integ/rest/RestResourceTest.java

```diff
@@ -323,6 +327,35 @@ class V1Beta1 {
         private static final String API_VERSION = "v1beta1";
 
         private final String requestApiPath = String.format("/api/%s/generations", API_VERSION);
+        private static final JsonFactory JSON_FACTORY = new JsonFactory();
+
+        boolean validJson(String jsonString) {
+            if (jsonString == null || jsonString.trim().isEmpty()) {
+                return false; // Test for empty
+            }
+
+            try (JsonParser parser = JSON_FACTORY.createParser(jsonString)) {
+
+                java.util.Set<String> seenKeys = new java.util.HashSet<>();
+                JsonToken token;
+
+                while ((token = parser.nextToken()) != null) {
+                    if (token == JsonToken.START_OBJECT) {
+                        seenKeys.clear();
+                    } else if (token == JsonToken.FIELD_NAME) {
+                        String key = parser.getCurrentName();
+                        if (!seenKeys.add(key)) {
+                            return false; // Duplicate key
+                        }
+                    } else if (token == JsonToken.END_OBJECT) {
+                        seenKeys.clear();
+                    }
+                }
+                return true;
+            } catch (Exception e) {
+                return false; // Invalid JSON
+            }
+        }
```

Comment:
I think you can greatly simplify this code:

```java
 private static final JsonFactory JSON_FACTORY = new JsonFactory().enable(JsonParser.Feature.STRICT_DUPLICATE_DETECTION);

        boolean isValidJson(String jsonString) {
            try (JsonParser parser = JSON_FACTORY.createParser(jsonString)) {
                return parser.readValueAsTree() != null;
            } catch (Exception e) {
                return false;
            }
        }
```

and do we want this kind of checking in the class itself as opposed to just the test?

Reply:
And I changed the name of the method to begin with "is" since it's a `boolean`.

Reply:
Yep much neater and great suggestion, I did have to add objectmapper as the readValueAsTree requires JsonFactory parser be initialized with an ObjectCodec.